challenge_4/java/zmiller91/src/zmiller91/InvertBinaryTree.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package zmiller91;
+
+/**
+ *
+ * @author zmiller
+ */
+public class InvertBinaryTree {
+    
+    /**
+     * Post order tree traversal to invert the tree from the bottom up.
+     * 
+     * @param root - binary tree
+     */
+    public static void invert(Node root) {
+        
+        // Stop at the n-1 row
```

Comment:
After further testing, the original logic works as intended. Unless you're seeing something I'm not, this works for: 

    * Single node trees
    * Balanced, complete binary trees
    * Balanced, incomplete binary trees
    * Unbalanced binary trees
    * Trees with at most 1 node

I'm curious as to what your thoughts are, though.